functions/common/config-common/src/main/java/org/springframework/cloud/fn/common/config/IntegrationComponentCustomizerBeanPostProcessor.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2022-2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.fn.common.config;
+
+import org.springframework.beans.BeansException;
+import org.springframework.beans.factory.BeanFactory;
+import org.springframework.beans.factory.BeanFactoryAware;
+import org.springframework.beans.factory.config.BeanPostProcessor;
+import org.springframework.core.ResolvableType;
+import org.springframework.integration.core.MessageProducer;
+import org.springframework.integration.core.MessageSource;
+import org.springframework.integration.dsl.IntegrationComponentSpec;
+import org.springframework.messaging.MessageHandler;
+
+/**
+ * The {@link BeanPostProcessor} to apply a {@link IntegrationComponentCustomizer} for a bean
+ * to process if a customizer for its type is present in the application context.
+ *
+ * @author Artem Bilan
+ *
+ * @since 1.2.1
+ */
+class IntegrationComponentCustomizerBeanPostProcessor implements BeanPostProcessor, BeanFactoryAware {
+
+	private BeanFactory beanFactory;
+
+	@Override
+	public void setBeanFactory(BeanFactory beanFactory) throws BeansException {
+		this.beanFactory = beanFactory;
+	}
+
+	@Override
+	public Object postProcessBeforeInitialization(Object bean, String beanName) throws BeansException {
+		if (bean instanceof MessageSource ||
+				bean instanceof MessageHandler ||
+				bean instanceof MessageProducer ||
+				bean instanceof IntegrationComponentSpec<?, ?>) {
+
+			ResolvableType integrationComponentCustomizerType =
+					ResolvableType.forClassWithGenerics(IntegrationComponentCustomizer.class, bean.getClass());
+
+			IntegrationComponentCustomizer<Object> beanCustomizer =
```

Comment:
This `postProcessBeforeInitialization()` is already heavy enough to check for extra bean for each bean initialized.
And this `c -> c` will cause extra object to be created on the heap and then removed by GC.

One day when lambdas became a trend, we decided with Gary do not use them in critical code pass: we lived without lambdas before and there is rally no reason to suffer performance just because it looks nicer to perform as lambda.

Reply:
While I do agree w/ avoiding them in hot paths (esp. tight loops) or even during frequent user operations, during config startup it is truly negligible and I enjoy using the lambda approach/style in these cases.

But at the end of the day, it can also come down to preference (perf aside) and not a big deal either way. 

Thanks for the discussion. I will resolve the comment.